sklearn/multioutput.py

```diff
@@ -0,0 +1,221 @@
+        within each target variable.
+        It does each target variable in y in series.
+
+    Attributes
+    __________
+    estimator : Sklearn estimator
+        The base estimator used to construct the model.
+
+    n_outputs : int
+        Number of outputs in the multi-output classifier
+
+    """
+
+    def __init__(self, estimator, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data.
+        Creates a seperate model for each Response column.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : (sparse) array-like, shape = [n_samples, n_outputs]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+
+        _check_estimator(self.estimator)
+        # check to see that the X and y have the same number of rows.
+        check_consistent_length(X, y)
+        if y.ndim == 1:
+            y = y.reshape(-1, 1)
+        # Calculate the number of classifiers
+        self.n_outputs = y.shape[1]
+        self.estimators_ = []
+        for i in range(self.n_outputs):
+            estimator = clone(self.estimator)   # make a fresh clone
+            self.estimators_.append(estimator.fit(X, y[:, i]))
+
+        return self
+
+    def predict(self, X):
+        """Predict multi-class multiple target variable using a model
+         trained for each target variable.
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        Returns
+        -------
+        y : (sparse) array-like, shape = [n_samples, n_outputs]
+            Multi-output targets. An indicator matrix turns on multilabel
+            classification.
+            Predicted multi-class targets across multiple predictors.
+            Note:  entirely separate models are generated
+            for each predictor.
+        """
+        check_is_fitted(self, 'estimators_')
+
+        results = np.column_stack([estimator.predict(X) for estimator in
+                                   self.estimators_])
+        return results
+
+    def predict_proba(self, X):
+        """Probability estimates. This returns prediction probabilites
+        for each class for each label.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+
+        Returns
+        -------
+        T : (sparse) array-like, shape = [n_samples, n_classes, n_outputs]
+            The class probabilities of the samples for each of the outputs
+        """
+        check_is_fitted(self, 'estimators_')
+
+        results = np.dstack([estimator.predict_proba(X) for estimator in
+                            self.estimators_])
+        return results
+
+    def score(self, X, y):
```

Comment:
Also the ``accuracy_score`` function seems to not accept ``multi output`` classification as of now since the ``type_of_target`` function classifies this type of output as ``multiclass-multioutput``. Could you please look into this and suggest what to do ?